src/components/ui/Suggest.jsx

```diff
@@ -0,0 +1,134 @@
+import React, { useEffect, useState } from 'react';
+import nconf from '@qwant/nconf-getter';
+import debounce from 'lodash.debounce';
+
+import SuggestsDropdown from 'src/components/ui/SuggestsDropdown';
+import { suggestResults } from 'src/adapters/suggest_sources';
+import { selectItem, modifyList } from 'src/libs/suggest';
+
+const geocoderConfig = nconf.get().services.geocoder;
+const SUGGEST_MAX_ITEMS = geocoderConfig.maxItems;
+const SUGGEST_USE_FOCUS = geocoderConfig.useFocus;
+const SUGGEST_FOCUS_MIN_ZOOM = 11;
+
+const Suggest = ({ withCategories, withGeoloc }) => {
+  const [items, setItems] = useState([]);
+  const [isOpen, setIsOpen] = useState(false);
+  const [lastQuery, setLastQuery] = useState('');
+  const [isPending, setIsPending] = useState('unset');
+  const [isPendingSubmit, setIsPendingSubmit] = useState(false);
+  const [isHighlighted, setIsHighlighted] = useState(false);
+  const searchInputDomHandler = document.getElementById('search');
+  let currentQuery = null;
+
+  useEffect(() => {
+    const handleFocus = () => setIsOpen(true);
+
+    const handleBlur = () => {
+      setIsHighlighted(false);
+      setIsOpen(false);
+    };
+
+    const handleInput = debounce(e => {
+      const typedValue = e.target.value;
+      setIsPending('fetching');
+
+      if (currentQuery) {
+        currentQuery.abort();
+      }
+
+      const query = suggestResults(typedValue, {
+        withCategories,
+        useFocus: SUGGEST_USE_FOCUS,
+        focusMinZoom: SUGGEST_FOCUS_MIN_ZOOM,
+        maxFavorites: !typedValue ? 5 : 2,
+        maxItems: SUGGEST_MAX_ITEMS,
+      });
+
+      currentQuery = query;
+      setLastQuery(typedValue);
+
+      query
+        .then(suggestions => modifyList(suggestions, withGeoloc))
+        .then(items => {
+          setItems(items);
+          setIsPending('completed');
+          currentQuery = null;
+        })
+        .catch(() => { /* Query aborted. Just ignore silently */ });
+    }, 100);
+
+    searchInputDomHandler.addEventListener('focus', handleFocus);
+    searchInputDomHandler.addEventListener('blur', handleBlur);
+    searchInputDomHandler.addEventListener('input', handleInput);
+
+    return () => {
+      searchInputDomHandler.removeEventListener('blur', handleFocus);
+      searchInputDomHandler.removeEventListener('blur', handleBlur);
+      searchInputDomHandler.removeEventListener('input', handleInput);
+    };
+  }, []);
+
+  useEffect(() => {
+    const handleSubmit = () => {
+      selectItem(items[0], true);
+    };
+
+    const handleKeyDown = async event => {
+      if (event.key === 'Esc' || event.key === 'Escape') { // esc
+        setIsOpen(false);
+        setIsHighlighted(false);
+      } else if (event.key === 'Enter' && !isHighlighted) {
```

Comment:
`SuggestsDropdown` watches for the 'Enter' key too. There is no reason to have both. I think this should be done in this component only, and the dropdown should now just manage the arrow keys and mouse interaction.

Another way is to merge `Suggest` and `SuggestsDropdown` in a single component, as they can't exist separately anyway.